Reconstruction/RecCalorimeter/src/components/SingleCaloTowerTool.cpp

```diff
@@ -43,6 +43,20 @@ StatusCode SingleCaloTowerTool::initialize() {
     error() << "There is no phi-eta segmentation." << endmsg;
     return StatusCode::FAILURE;
   }
+  // Take readout bitfield decoder from GeoSvc
+  m_decoder =
+      std::shared_ptr<DD4hep::DDSegmentation::BitField64>(m_geoSvc->lcdd()->readout(m_readoutName).idSpec().decoder());
+  // check if decoder contains "layer"
+  std::vector<std::string> fields;
+  for (uint itField = 0; itField < m_decoder->size(); itField++) {
+    fields.push_back((*m_decoder)[itField].name());
+  }
+  auto iter = std::find(fields.begin(), fields.end(), "layer");
+  if (iter == fields.end()) {
```

Comment:
You can't decide about adding 'layer restriction' based on the existence of the field in the readout. What if someone desn't need it?